ui_framework/components/prompt_for_items/prompt_for_items.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import classnames from 'classnames';
+
+import { KuiButtonIcon, KuiLinkButton } from '../button';
+import { KuiPromptForItemsMessage } from  './prompt_for_items_message';
+
+export function KuiPromptForItems({ itemType, promptMessage, promptButtonText, addHref, className, ...rest }) {
+  const classes = classnames('kuiPromptForItems', className);
+  return (
+    <div className={classes} {...rest} >
+      <KuiPromptForItemsMessage>
+        {promptMessage}
+      </KuiPromptForItemsMessage>
+
+      <div className="kuiPromptForItems__actions">
+        <KuiLinkButton
+          icon={<KuiButtonIcon type="create" />}
+          aria-label={`Add a new ${itemType}`}
+          className="testClass1 testClass2"
+          data-test-subj={`addNewPromptButton`}
+          buttonType="primary"
+          href={addHref}
+        >
```

Comment:
Sorry for chiming in so late. Even though I agree with the preference of composition via the `children` prop, this looks like a good compromise here. :+1: